plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -0,0 +1,385 @@
+import org.apache.iceberg.catalog.SessionCatalog;
+import org.apache.iceberg.catalog.SessionCatalog.SessionContext;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.RESTException;
+import org.apache.iceberg.rest.RESTSessionCatalog;
+import org.apache.iceberg.rest.auth.OAuth2Properties;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.hive.HiveMetadata.TABLE_COMMENT;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_CATALOG_ERROR;
+import static io.trino.plugin.iceberg.IcebergTableProperties.LOCATION_PROPERTY;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static java.lang.String.format;
+import static java.util.UUID.randomUUID;
+
+public class TrinoRestCatalog
+        implements TrinoCatalog
+{
+    private final RESTSessionCatalog sessionCatalog;
+    private final CatalogName catalogName;
+    private final String trinoVersion;
+    private final boolean useUniqueTableLocation;
+
+    private final ConcurrentHashMap<String, Table> tableCache = new ConcurrentHashMap<>();
+
+    public TrinoRestCatalog(
+            RESTSessionCatalog sessionCatalog,
+            CatalogName catalogName,
+            String trinoVersion,
+            boolean useUniqueTableLocation)
+    {
+        this.sessionCatalog = sessionCatalog;
+        this.catalogName = catalogName;
+        this.trinoVersion = trinoVersion;
+        this.useUniqueTableLocation = useUniqueTableLocation;
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return sessionCatalog.listNamespaces(convert(session)).stream()
+                .map(Namespace::toString)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        sessionCatalog.dropNamespace(convert(session), Namespace.of(namespace));
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return ImmutableMap.copyOf(sessionCatalog.loadNamespaceMetadata(convert(session), Namespace.of(namespace)));
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        checkArgument(!properties.containsKey(LOCATION_PROPERTY), "Database cannot be created with a location set");
```

Comment:
Why is that?

Reply:
This was caused by an issue in the backing JdbcCatalog used for testing.  I've removed this test and will update the tests to reflect that along with a note to update the tests after the Iceberg issue is addressed.  The problem was that the catalog implementation was always using the default location even if it was explicitly set, so the tests for setting location would not pass.

Reply:
Upstream issue is: https://github.com/apache/iceberg/pull/5737